tests/integration/test_charm.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from typing import Any
+from pathlib import Path
+
+import yaml
+import pytest
+import lightkube
+from lightkube.generic_resource import create_namespaced_resource
+
+from pytest_operator.plugin import OpsTest
+
+log = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = "training-operator"
+
+
+def lightkube_create_ns_resource(kind: str) -> Any:
+    job_class = create_namespaced_resource(
+        group="kubeflow.org", version="v1", kind=kind
+    )
+    return job_class
+
+
+def create_training_job(manifest_file: str, namespace: str) -> bool:
+    lightkube_client = lightkube.Client()
+    manifest_path = Path.cwd() / "examples" / manifest_file
+    kubernetes_crd = lightkube.codecs.load_all_yaml(Path.read_text(manifest_path))[0]
+    try:
+        lightkube_client.create(kubernetes_crd)
+    except ApiError as e:
+        log.error(
+            f"Caught {e.status} while trying to create {kubernetes_crd.metadata.name}"
+        )
+        log.error(traceback.format_exc())
+        return False
+    else:
+        return True
+
+
+def get_training_job_status(name: str, namespace: str, kind: str) -> dict:
+    lightkube_client = lightkube.Client()
+    job_class = lightkube_create_ns_resource(kind=kind)
+    job = lightkube_client.get(tfjob_class, name=name, namespace=namespace)
+    return job.get("status")
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Build the charm and deploy it with trust=True.
+
+    Assert on the unit status.
+    """
+    charm_under_test = await ops_test.build_charm(".")
+    image_path = METADATA["resources"]["training-operator-image"]["upstream-source"]
+    resources = {"training-operator-image": image_path}
+
+    await ops_test.model.deploy(
+        charm_under_test, resources=resources, application_name=APP_NAME, trust=True
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=60 * 10
+    )
+    assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
+
+
+@pytest.mark.abort_on_fail
```

Comment:
yeah I think the idea of that since pytest typically runs tests independently, the annotation lets you say "well nothing else can pass if this fails so I should abort if this fails" about a test(s).  It could be used on anything, but really it is just used on tests that you want to exit after if they fail (so deployment is a good one, but if you've got 3 tests that all work independently then probably you want want it as you want to know if they each passed/failed every time)